src/app/container/launch/launch.component.html

```diff
@@ -30,7 +30,7 @@
       <div class="row">
         <div class="col-sm-4">
           <mat-form-field>
-            <mat-select name="toolVersions" [(value)]="currentDescriptor" (selectionChange)="changeDescriptor($event)">
```

Comment:
1. There is already an `ngOnChanges` that handles the change.
2. The `changeDescriptor()` method being called takes 3 parameters, not the one that was being passed to it.